experiments_main.cpp

```diff
@@ -107,119 +111,114 @@ DEFINE_uint32(num_threads,
 
 using namespace ts;
 
-template <class char_type, class kmer_type, class embed_type>
-struct SeqGenModule {
-    SeqGenModule()
-        : min_hash(int_pow<uint32_t>(FLAGS_alphabet_size, FLAGS_kmer_size),
-                   FLAGS_embed_dim,
-                   parse_hash_algorithm(FLAGS_hash_alg)),
-          wmin_hash(int_pow<uint32_t>(FLAGS_alphabet_size, FLAGS_kmer_size),
-                    FLAGS_embed_dim,
-                    FLAGS_max_len,
-                    parse_hash_algorithm(FLAGS_hash_alg)),
-          omin_hash(int_pow<uint32_t>(FLAGS_alphabet_size, FLAGS_kmer_size),
-                    FLAGS_embed_dim,
-                    FLAGS_max_len,
-                    FLAGS_tuple_length,
-                    parse_hash_algorithm(FLAGS_hash_alg)),
-          tensor_sketch(FLAGS_alphabet_size, FLAGS_embed_dim, FLAGS_tuple_length),
-          tensor_slide(FLAGS_alphabet_size,
-                       ceil(sqrt(FLAGS_embed_dim)),
-                       FLAGS_tuple_length,
-                       FLAGS_window_size,
-                       FLAGS_stride),
-          l1SketchBin32(FLAGS_embed_dim, ceil(sqrt(FLAGS_embed_dim)), FLAGS_seq_len),
-          l1Sketch(FLAGS_embed_dim, ceil(sqrt(FLAGS_embed_dim)), FLAGS_seq_len) {}
+template <class char_type, class kmer_type, class embed_type, class... SketchAlgorithms>
+class ExperimentRunner {
+    static constexpr size_t NumAlgorithms = sizeof...(SketchAlgorithms);
+    std::tuple<SketchAlgorithms...> algorithms_;
```

Comment:
The style in this project doesn't use underscores at the end of local variables or private members.